Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -0,0 +1,1097 @@
+#include "WarpX.H"
+#include <AMReX_Scan.H>
+#include <AMReX_iMultiFab.H>
+#include <AMReX_MultiFab.H>
+
+/**
+* \brief auxiliary function to count the amount of faces which still need to be extended
+*/
+amrex::Array1D<int, 0, 2>
+WarpX::CountExtFaces() {
+    amrex::Array1D<int, 0, 2> sums{0, 0, 0};
+#ifdef AMREX_USE_EB
+    for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+        int sum = 0;
+        for (amrex::MFIter mfi(*m_flag_ext_face[maxLevel()][idim]); mfi.isValid(); ++mfi) {
+            amrex::Box const &box = mfi.validbox();
+            auto const &flag_ext_face = m_flag_ext_face[maxLevel()][idim]->array(mfi);
+            amrex::ParallelFor(box, [=, &sum](int i, int j, int k) {
```

Comment:
Shall we add this, too? Cc: @WeiqunZhang 
```suggestion
            amrex::ParallelFor(box, [=, &sum] AMREX_GPU_DEVICE (int i, int j, int k) {
```